libs/wire-api/src/Wire/API/MLS/Serialisation.hs

```diff
@@ -81,6 +82,16 @@ parseMLSVector getItem = do
       pos <- bytesRead
       (:) x <$> (if pos < endPos then go endPos else pure [])
 
+serialiseMLSVector ::
+  forall w a.
+  (Binary w, Integral w) =>
+  (a -> Put) ->
+  [a] ->
+  Put
+serialiseMLSVector p as = do
+  put @w . fromIntegral . length $ as
```

Comment:
This is not correct. Vectors begin with the whole size of their binary representation, not the number of items. See https://www.rfc-editor.org/rfc/rfc5246#section-4.3.

Reply:
Thanks for spotting! Fixed in [cc6548a](https://github.com/wireapp/wire-server/pull/2628/commits/cc6548a051b3ba565478cf6bdbd8674da9f33a6d).